File: daily/2026/3567_Minimum_Absolute_Difference_in_Sliding_Submatrix.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;

class Solution {
public:
    vector<vector<int>> minAbsDiff(vector<vector<int>>& grid, int k) {
        int m = grid.size();
        int n = grid[0].size();
        vector<vector<int>> ans(m - k + 1, vector<int>(n - k + 1, 0));
        for (int i = 0; i + k - 1 < m; i++){
            for (int j = 0; j + k - 1 < n; j++){
                set<int> curr;
                for (int x = i; x < i + k; x++){
                    for (int y = j; y < j + k; y++){
                        curr.insert(grid[x][y]);
                    }
                }
                if (curr.size() == 1){
                    ans[i][j] = 0;
                    continue;
                }
                int d = INT_MAX;
                for (auto it = curr.begin(); it != prev(curr.end()); it++){
                    auto nxt = next(it);
                    d = min(d, abs(*nxt - *it));
                }
                ans[i][j] = d;
            }
        }
        return ans;
    }
};

int main(){
    vector<vector<int>> grid = {{1,8},{3,-2}};
    int k = 2;
    // vector<vector<int>> grid = {{3,-1}};
    // int k = 1;
    // vector<vector<int>> grid = {{1,-2,3},{2,3,5}};
    // int k = 2;
    Solution S;
    vector<vector<int>> ans = S.minAbsDiff(grid, k);
    for (auto& row : ans){
        for (int col : row){
            cout << col << " ";
        }
        cout << endl;
    }
    return 0;
}

File: daily_python/3567_Minimum_Absolute_Difference_in_Sliding_Submatrix.py
import os
import sys
from typing import List

class Solution:
    def minAbsDiff(self, grid: List[List[int]], k: int) -> List[List[int]]:
        m, n = len(grid), len(grid[0])
        ans = [[0] * (n - k + 1) for _ in range(m - k + 1)]
        for i in range(m - k + 1):
            for j in range(n - k + 1):
                res = []
                for x in range(i, i + k):
                    for y in range(j, j + k):
                        res.append(grid[x][y])
                res.sort()
                kmin = float('inf')
                for ptr in range(1, len(res)):
                    if res[ptr-1] != res[ptr]:
                        kmin = min(kmin, abs(res[ptr] - res[ptr-1]))
                if kmin != float('inf'):
                    ans[i][j] = kmin
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            grid = [list(map(int, row.split(','))) for row in lines[0].split('=')[1].strip()[2:-3].split('},{')]
            k = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((grid, k))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for grid, k in dataset:
        results.append(solution.minAbsDiff(grid, k))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')